blockchain/message_pool/src/msgpool.rs

```diff
@@ -794,15 +908,29 @@ pub mod test_provider {
             Ok(Cid::default())
         }
 
-        fn state_get_actor(&self, addr: &Address, _ts: &Tipset) -> Result<ActorState, Errors> {
-            let s = self.state_sequence.get(addr);
-            let mut sequence = 0;
-            if let Some(sq) = s {
-                sequence = *sq;
+        fn get_actor_after(&self, addr: &Address, ts: &Tipset) -> Result<ActorState, Errors> {
+            let mut msgs: Vec<SignedMessage> = Vec::new();
+            for b in ts.blocks() {
+                if let Some(ms) = self.bmsgs.get(b.cid()) {
+                    for m in ms {
+                        if m.from() == addr {
+                            msgs.push(m.clone());
+                        }
+                    }
+                }
+            }
+            msgs.sort_by_key(|m| m.sequence());
```

Comment:
I don't know how much this would help but if implemented an ordering trait by sequecnce on the signed message and directly pushed into a BtreeSet you could get a n(log2n). This gives us a n + nlog2n because it has to iterate over blocks and then sort them. But it is a bit pendatic on my part

Reply:
ah i see. that'd be cool, but this is for tests, so i'd err on the side of just getting this in

Reply:
makes sense!